ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/InMemory.hs

```diff
@@ -1,613 +1 @@
-
-{-------------------------------------------------------------------------------
-  Support for testing
--------------------------------------------------------------------------------}
-
-pureBlock :: blk -> Ap m l blk ()
-pureBlock = ReapplyVal
-
-ledgerDbPush' :: ApplyBlock l blk
-              => LedgerDbCfg l -> blk -> LedgerDB l -> LedgerDB l
-ledgerDbPush' cfg b = runIdentity . ledgerDbPush cfg (pureBlock b)
-
-ledgerDbPushMany' :: ApplyBlock l blk
-                  => LedgerDbCfg l -> [blk] -> LedgerDB l -> LedgerDB l
-ledgerDbPushMany' cfg bs =
-  runIdentity . ledgerDbPushMany (const $ pure ()) cfg (map pureBlock bs)
-
-ledgerDbSwitch' :: forall l blk. ApplyBlock l blk
-                => LedgerDbCfg l
-                -> Word64 -> [blk] -> LedgerDB l -> Maybe (LedgerDB l)
-ledgerDbSwitch' cfg n bs db =
-    case runIdentity $ ledgerDbSwitch cfg n (const $ pure ()) (map pureBlock bs) db of
-      Left  ExceededRollback{} -> Nothing
-      Right db'                -> Just db'
-
-{-------------------------------------------------------------------------------
-  Serialisation
--------------------------------------------------------------------------------}
-
--- | Version 1: uses versioning ('Ouroboros.Consensus.Util.Versioned') and only
--- encodes the ledger state @l@.
-snapshotEncodingVersion1 :: VersionNumber
-snapshotEncodingVersion1 = 1
-
--- | Encoder to be used in combination with 'decodeSnapshotBackwardsCompatible'.
-encodeSnapshot :: (l -> Encoding) -> l -> Encoding
-encodeSnapshot encodeLedger l =
-    encodeVersion snapshotEncodingVersion1 (encodeLedger l)
-
--- | To remain backwards compatible with existing snapshots stored on disk, we
--- must accept the old format as well as the new format.
---
--- The old format:
--- * The tip: @WithOrigin (RealPoint blk)@
--- * The chain length: @Word64@
--- * The ledger state: @l@
---
--- The new format is described by 'snapshotEncodingVersion1'.
---
--- This decoder will accept and ignore them. The encoder ('encodeSnapshot') will
--- no longer encode them.
-decodeSnapshotBackwardsCompatible ::
-     forall l blk.
-     Proxy blk
-  -> (forall s. Decoder s l)
-  -> (forall s. Decoder s (HeaderHash blk))
-  -> forall s. Decoder s l
-decodeSnapshotBackwardsCompatible _ decodeLedger decodeHash =
-    decodeVersionWithHook
-      decodeOldFormat
-      [(snapshotEncodingVersion1, Decode decodeVersion1)]
-  where
-    decodeVersion1 :: forall s. Decoder s l
-    decodeVersion1 = decodeLedger
-
-    decodeOldFormat :: Maybe Int -> forall s. Decoder s l
-    decodeOldFormat (Just 3) = do
-        _ <- withOriginRealPointToPoint <$>
-               decodeWithOrigin (decodeRealPoint @blk decodeHash)
-        _ <- Dec.decodeWord64
-        decodeLedger
-    decodeOldFormat mbListLen =
-        fail $
-          "decodeSnapshotBackwardsCompatible: invalid start " <>
-          show mbListLen
+module Ouroboros.Consensus.Storage.LedgerDB.InMemory () where
```

Comment:
Should this instead expose the whole LedgerDB module so that clients code wont break just yet?